src/stats/mod.rs

```diff
@@ -1 +1,300 @@
+use crate::data_channel::data_channel_state::RTCDataChannelState;
+use crate::data_channel::RTCDataChannel;
+use crate::dtls_transport::dtls_fingerprint::RTCDtlsFingerprint;
+use crate::peer_connection::certificate::RTCCertificate;
+use crate::rtp_transceiver::rtp_codec::RTCRtpCodecParameters;
+use crate::rtp_transceiver::PayloadType;
+use crate::sctp_transport::RTCSctpTransport;
 
+use ice::agent::agent_stats::{CandidatePairStats, CandidateStats};
+use ice::candidate::{CandidatePairState, CandidateType};
+use ice::network_type::NetworkType;
+use stats_collector::StatsCollector;
+
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use tokio::time::Instant;
+
+pub mod stats_collector;
+
+pub enum SourceStatsType {
+    CandidatePair(CandidatePairStats),
+    LocalCandidate(CandidateStats),
+    RemoteCandidate(CandidateStats),
+}
+
+#[derive(Debug)]
+pub enum StatsReportType {
+    CandidatePair(ICECandidatePairStats),
+    CertificateStats(CertificateStats),
+    Codec(CodecStats),
+    DataChannel(DataChannelStats),
+    LocalCandidate(ICECandidateStats),
+    PeerConnection(PeerConnectionStats),
+    RemoteCandidate(ICECandidateStats),
+    SCTPTransport(ICETransportStats),
+    Transport(ICETransportStats),
+}
+
+impl From<SourceStatsType> for StatsReportType {
+    fn from(stats: SourceStatsType) -> Self {
+        match stats {
+            SourceStatsType::CandidatePair(stats) => StatsReportType::CandidatePair(stats.into()),
+            SourceStatsType::LocalCandidate(stats) => StatsReportType::LocalCandidate(stats.into()),
+            SourceStatsType::RemoteCandidate(stats) => {
+                StatsReportType::RemoteCandidate(stats.into())
+            }
+        }
+    }
+}
+
+// TODO: should this be some form of String-indexed HashMap?
+#[derive(Debug)]
+pub struct StatsReport {
+    pub(crate) reports: Vec<StatsReportType>,
+}
```

Comment:
I think that the Pion `StatsReport` is actually a type alias to something like a Rust `std::collection::HashMap`. I thought that it might make sense to have a struct, and then convert types when serializing to JSON (which I'm not completely sure how to do).

Does that make sense? Maybe rather than reports being a `Vec<StatsReportType>` it should be a `HashMap<String, StatsReportType>`?